File: diversity/go/src/bftsmartclient/bftsmartclient.go
package bftsmartclient

import "bftsmart"
                                            
func CreateServiceProxy(id int, clspath string) int {             
	bftsmart.SetClasspath(clspath)
	bftsmart.CarregarJvm()
	return bftsmart.CreateServiceProxy(id)
}                                                 
                                                  
func InvokeOrdered(command []byte) []byte {       
        return bftsmart.InvokeOrdered(command)
}
                                                  
func InvokeUnordered(command []byte) []byte {     
        return bftsmart.InvokeUnordered(command)
}                                                 
                                                  
func FinalizarJvm() {                             
	bftsmart.FinalizarJvm()
}                                                 

File: diversity/cpp/bftsmart-client.cpp
#include "bftsmart-client.hpp"

bftsmart::BftSmartClient::BftSmartClient(int id, string classpath)
{
	setClasspath(classpath.c_str());
	carregarJvm();
	createServiceProxy(id);
}

int bftsmart::BftSmartClient::invokeOrdered(BYTE request[], int size, BYTE response[])
{
	return invokeOrdered(request, size, response);
}

int bftsmart::BftSmartClient::invokeUnordered(BYTE request[], int size, BYTE response[])
{
	return invokeUnordered(request, size, response);
}

bftsmart::BftSmartClient::~BftSmartClient()
{
	finalizarJvm();
}
